doc/man7/provider-base.pod

```diff
@@ -10,165 +10,193 @@ provider-base
- CRYPTO_malloc                  OSSL_FUNC_CRYPTO_MALLOC
- CRYPTO_zalloc                  OSSL_FUNC_CRYPTO_ZALLOC
- CRYPTO_memdup                  OSSL_FUNC_CRYPTO_MEMDUP
- CRYPTO_strdup                  OSSL_FUNC_CRYPTO_STRDUP
- CRYPTO_strndup                 OSSL_FUNC_CRYPTO_STRNDUP
- CRYPTO_free                    OSSL_FUNC_CRYPTO_FREE
- CRYPTO_clear_free              OSSL_FUNC_CRYPTO_CLEAR_FREE
- CRYPTO_realloc                 OSSL_FUNC_CRYPTO_REALLOC
- CRYPTO_clear_realloc           OSSL_FUNC_CRYPTO_CLEAR_REALLOC
- CRYPTO_secure_malloc           OSSL_FUNC_CRYPTO_SECURE_MALLOC
- CRYPTO_secure_zalloc           OSSL_FUNC_CRYPTO_SECURE_ZALLOC
- CRYPTO_secure_free             OSSL_FUNC_CRYPTO_SECURE_FREE
- CRYPTO_secure_clear_free       OSSL_FUNC_CRYPTO_SECURE_CLEAR_FREE
- CRYPTO_secure_allocated        OSSL_FUNC_CRYPTO_SECURE_ALLOCATED
- BIO_new_file                   OSSL_FUNC_BIO_NEW_FILE
- BIO_new_mem_buf                OSSL_FUNC_BIO_NEW_MEMBUF
- BIO_read_ex                    OSSL_FUNC_BIO_READ_EX
- BIO_free                       OSSL_FUNC_BIO_FREE
- BIO_vprintf                    OSSL_FUNC_BIO_VPRINTF
- OPENSSL_cleanse                OSSL_FUNC_OPENSSL_CLEANSE
- OSSL_SELF_TEST_set_callback    OSSL_FUNC_SELF_TEST_CB
+ {core_gettable_params}           OSSL_FUNC_CORE_GETTABLE_PARAMS
+ {core_get_params}                OSSL_FUNC_CORE_GET_PARAMS
+ {core_thread_start}              OSSL_FUNC_CORE_THREAD_START
+ {core_get_library_context}       OSSL_FUNC_CORE_GET_LIBRARY_CONTEXT
+ {core_new_error}                 OSSL_FUNC_CORE_NEW_ERROR
+ {core_set_error_debug}           OSSL_FUNC_CORE_SET_ERROR_DEBUG
+ {core_set_error}                 OSSL_FUNC_CORE_SET_ERROR
+ {CRYPTO_malloc}                  OSSL_FUNC_CRYPTO_MALLOC
+ {CRYPTO_zalloc}                  OSSL_FUNC_CRYPTO_ZALLOC
+ {CRYPTO_memdup}                  OSSL_FUNC_CRYPTO_MEMDUP
+ {CRYPTO_strdup}                  OSSL_FUNC_CRYPTO_STRDUP
+ {CRYPTO_strndup}                 OSSL_FUNC_CRYPTO_STRNDUP
+ {CRYPTO_free}                    OSSL_FUNC_CRYPTO_FREE
+ {CRYPTO_clear_free}              OSSL_FUNC_CRYPTO_CLEAR_FREE
+ {CRYPTO_realloc}                 OSSL_FUNC_CRYPTO_REALLOC
+ {CRYPTO_clear_realloc}           OSSL_FUNC_CRYPTO_CLEAR_REALLOC
+ {CRYPTO_secure_malloc}           OSSL_FUNC_CRYPTO_SECURE_MALLOC
+ {CRYPTO_secure_zalloc}           OSSL_FUNC_CRYPTO_SECURE_ZALLOC
+ {CRYPTO_secure_free}             OSSL_FUNC_CRYPTO_SECURE_FREE
+ {CRYPTO_secure_clear_free}       OSSL_FUNC_CRYPTO_SECURE_CLEAR_FREE
+ {CRYPTO_secure_allocated}        OSSL_FUNC_CRYPTO_SECURE_ALLOCATED
+ {BIO_new_file}                   OSSL_FUNC_BIO_NEW_FILE
+ {BIO_new_mem_buf}                OSSL_FUNC_BIO_NEW_MEMBUF
+ {BIO_read_ex}                    OSSL_FUNC_BIO_READ_EX
+ {BIO_free}                       OSSL_FUNC_BIO_FREE
+ {BIO_vprintf}                    OSSL_FUNC_BIO_VPRINTF
+ {OPENSSL_cleanse}                OSSL_FUNC_OPENSSL_CLEANSE
+ {OSSL_SELF_TEST_set_callback}    OSSL_FUNC_SELF_TEST_CB
```

Comment:
This is an error: instead of the exported symbol `OSSL_SELF_TEST_set_callback` this line should contain the name of the dispatched function:

```
 {self_set_cb}         OSSL_FUNC_SELF_TEST_CB
```

Reply:
I reverted the renaming of the self test callback for the moment. The names need some straightening out, but I'll keep that for another pr.